plugins/main/public/components/common/modules/modules-helper.js

```diff
@@ -182,4 +184,59 @@ If this case happens, the implicit filters are regenerated and the function is c
 
     return filters;
   }
+
+  /**
+   *
+   * @param {string} indexPattern - Index pattern id being used
+   * @param {Filter[] | undefined} filters - (Optional) Filter Array in which to search if a pinned agent exists
+   * @returns
+   */
+  static getImplicitPinnedAgent(indexPattern, filters = []) {
+    const filterManager = useFilterManager().filterManager;
+    const initialFilters =
+      filters?.length > 0 ? filters : filterManager.getFilters();
+    const pinnedAgentByFilterManager = initialFilters.find(
+      filter =>
+        filter?.meta?.key === 'agent.id' && !!filter?.$state?.isImplicit,
+    );
+    const url = window.location.href;
+    const regex = new RegExp('agentId=' + '[^&]*');
+    const match = url.match(regex);
+    const isPinnedAgentByUrl = match && match[0];
+    if (pinnedAgentByFilterManager && isPinnedAgentByUrl) {
+      return {
+        ...pinnedAgentByFilterManager,
+        meta: {
+          ...pinnedAgentByFilterManager.meta,
+          index: indexPattern,
+        },
+        $state: { store: FilterStateStore.APP_STATE, isImplicit: true },
+      };
+    }
+
+    if (isPinnedAgentByUrl) {
+      const agentId = match[0].split('=')[1];
+      return {
+        meta: {
+          alias: null,
+          disabled: false,
+          key: 'agent.id',
+          negate: false,
+          params: { query: agentId },
+          type: 'phrase',
+          index: indexPattern,
+        },
+        query: {
+          match: {
+            'agent.id': {
+              query: agentId,
+              type: 'phrase',
+            },
+          },
+        },
+        $state: { store: 'appState', isImplicit: true },
+      };
+    }
+    return undefined;
+  }
```

Comment:
is it still necessary with the dataSource?

Reply:
Removed